R/SummaryPanel4a.R

```diff
@@ -0,0 +1,67 @@
+#' Module UI for the covariate summary plot tab.
+#' 
+#' @param id ID of the module
+#' @return Tab 4a Summary
+metaregression_summary_panel_ui <- function(id) {
+  ns <- NS(id)
+  fluidPage(
+    h4('Summary Characteristic Plot'),
+    uiOutput(ns('toggle_covariate_baseline')),
+    plotOutput(outputId = ns('covariate_plot')),
+    textOutput(outputId = ns('covariate_info')),
+    radioButtons(inputId = ns('format_covariate_plot'), 
+                 label = 'Document format', 
+                 choices = c('PDF' = 'pdf', 'PNG' = 'png'), 
+                 inline = TRUE),
+    downloadButton(outputId = ns('download_covariate_summary'))
+  )
+}
+
+#' Module server for the covariate summary plot tab.
+#'
+#' @param id ID of the module
+#' @param all_data Study data including covariate columns, in wide or long format
+#' @param metaoutcome Reactive containing meta analysis outcome: 'Continuous' or 'Binary'
+#' @param treatment_df Reactive containing data frame containing treatment IDs (Number) and names (Label)
+#' @return Covariate summary plot tab
+#' 
+metaregression_summary_panel_server <- function(id, all_data, metaoutcome, treatment_df) {
+  moduleServer(id, function(input, output, session) {
+    
+    # Toggle between covariate and baseline risk, only when there is a covariate
+    output$toggle_covariate_baseline <- renderUI({
+      # If there is a covariate (i.e. the covariate name is not NA)
+      if(!is.na(FindCovariateNames(all_data())[1])) {
+        shinyWidgets::radioGroupButtons(
+              inputId = session$ns('toggle_covariate_baseline'),
+              choices = c('Covariate', 'Baseline risk'),
+              status = 'primary'
+        )
+      }
+    })
+    
+    # Render covariate summary plot
+    observe({
```

Comment:
The `observe()` wrapper is not needed here. The `renderPlot()` call takes care of all of the reactivity.

Reply:
There's a fatal error without it, because of the width = ... argument.

Reply:
Ah ok, that's a shame. Can you add a comment explaining this please?